addons/Request/Request.lua

```diff
@@ -0,0 +1,305 @@
+--[[
+Copyright © 2015, Selindrile
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Request nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL Selindrile BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+]]
+
+require('luau')
+
+_addon.name = 'Request'
+_addon.author = 'Selindrile'
+_addon.commands = {'request','rq'}
+_addon.version = 1.1
+_addon.language = 'english'
+
+defaults = T{}
+defaults.mode = 'whitelist'
+defaults.whitelist = S{}
+defaults.blacklist = S{}
+defaults.nicknames = S{}
+defaults.forbidden = S{}
+defaults.partylock = 'On'
+defaults.exactlock = 'On'
+defaults.requestlock = 'Off'
```

Comment:
I thought you meant for what was contained within partylock/exactlock/requestlock, so that it would show up capitalized when a user viewed the settings with //request, but I see you asked me to use true or false there. Did you mean for me to pascalcase: defaults.PartyLock, defaults.ExactLock, defaults.RequestLock like so?